PPA build failures

(Following the format of a bug filed in 2018, it might even be the same problem: https://github.com/OpenRCT2/OpenRCT2/issues/7468)

The nightly PPA build is failing.  Included is a snippet from the end of the build log (https://launchpadlibrarian.net/535195924/buildlog_ubuntu-focal-amd64.openrct2_0.3.3-develop-0~2459-202104240229-g1aec32056~ubuntu20.04.1_BUILDING.txt.gz):
```
Added: /<<PKGBUILDDIR>>/resources/g2/track/raptor/small_steep_to_flat_up_diag_lift_2.png
Added: /<<PKGBUILDDIR>>/resources/g2/track/raptor/small_steep_to_flat_up_diag_lift_3.png
Added: /<<PKGBUILDDIR>>/resources/g2/track/raptor/small_steep_to_flat_up_diag_lift_4.png
Finished
make[5]: Leaving directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu'
[100%] Built target g2
make[4]: Leaving directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu'
/usr/bin/cmake -E cmake_progress_start "/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/CMakeFiles" 0
make[3]: Leaving directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu'
make[2]: Leaving directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu'
Downloading /<<PKGBUILDDIR>>/debian/openrct2/usr/share/openrct2/sequence/
CMake Error at /<<PKGBUILDDIR>>/cmake/download.cmake:23 (file):
  file DOWNLOAD HASH mismatch

    for file: [/<<PKGBUILDDIR>>/debian/openrct2/usr/share/openrct2/sequence//title-sequences.zip]
      expected hash: [304d13a126c15bf2c86ff13b81a2f2cc1856ac8d]
        actual hash: [da39a3ee5e6b4b0d3255bfef95601890afd80709]
             status: [6;"Couldn't resolve host name"]

Call Stack (most recent call first):
  cmake_install.cmake:47 (download_openrct2_zip)


make[1]: *** [Makefile:77: install] Error 1
make[1]: Leaving directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu'
dh_auto_install: error: cd obj-x86_64-linux-gnu && make -j1 install DESTDIR=/<<BUILDDIR>>/openrct2-0.3.3-develop-0\~2459-202104240229/debian/openrct2 AM_UPDATE_INFO_DIR=no returned exit code 2
make: *** [debian/rules:6: binary] Error 25
dpkg-buildpackage: error: fakeroot debian/rules binary subprocess returned exit status 2
--------------------------------------------------------------------------------
Build finished at 2021-04-24T03:01:59Z

Finished
--------


+------------------------------------------------------------------------------+
| Cleanup                                                                      |
+------------------------------------------------------------------------------+

Purging /<<BUILDDIR>>
Not removing build depends: as requested
E: Build failure (dpkg-buildpackage died)
```
https://launchpad.net/%7Eopenrct2/+archive/ubuntu/nightly

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -401,22 +401,28 @@ if (NOT MACOS_BUNDLE OR (MACOS_BUNDLE AND WITH_TESTS))
     install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" --build \"${CMAKE_CURRENT_BINARY_DIR}\" --target g2)")
     if (DOWNLOAD_TITLE_SEQUENCES)
         # Checks if this version of the title sequences are already installed, updates if necessary
+        # if /data/sequence exists, it will overwrite this anyways (see `install(DIRECTORY "data/" ...)` below)
+        # therefore, the existence of that directory will skip this download
         install(CODE "
             include(${ROOT_DIR}/cmake/download.cmake)
             download_openrct2_zip(
                 ZIP_VERSION ${TITLE_SEQUENCE_VERSION}
                 DOWNLOAD_DIR \$ENV{DESTDIR}${CMAKE_INSTALL_FULL_DATADIR}/${PROJECT_NAME}/sequence/
+                SKIP_IF_EXISTS ${CMAKE_SOURCE_DIR}/data/sequence/
                 ZIP_URL ${TITLE_SEQUENCE_URL}
                 SHA1 ${TITLE_SEQUENCE_SHA1}
             )")
     endif ()
     if (DOWNLOAD_OBJECTS)
         # Checks if this version of the objects are already installed, updates if necessary
+        # if /data/object exists, it will overwrite this anyways (see `install(DIRECTORY "data/" ...)` below)
+        # therefore, the existence of that directory will skip this download
         install(CODE "
             include(${ROOT_DIR}/cmake/download.cmake)
             download_openrct2_zip(
                 ZIP_VERSION ${OBJECTS_VERSION}
                 DOWNLOAD_DIR \$ENV{DESTDIR}${CMAKE_INSTALL_FULL_DATADIR}/${PROJECT_NAME}/object/
+                SKIP_IF_EXISTS ${CMAKE_SOURCE_DIR}/data/object/
                 ZIP_URL ${OBJECTS_URL}
                 SHA1 ${OBJECTS_SHA1}
             )")

--- a/cmake/download.cmake
+++ b/cmake/download.cmake
@@ -1,5 +1,6 @@
 function(download_openrct2_zip)
     set(oneValueArgs ZIP_VERSION DOWNLOAD_DIR ZIP_URL SHA1)
+    set(multiValueArgs SKIP_IF_EXISTS)
     cmake_parse_arguments(DOWNLOAD_OPENRCT2 "${options}" "${oneValueArgs}"
                         "${multiValueArgs}" ${ARGN} )
 
@@ -15,9 +16,18 @@ function(download_openrct2_zip)
                 set(DOWNLOAD_ZIP 1)
             endif ()
         else ()
-        set(DOWNLOAD_ZIP 1)
+            set(DOWNLOAD_ZIP 1)
         endif ()
     endif ()
+
+    foreach(check_exist_dir ${DOWNLOAD_OPENRCT2_SKIP_IF_EXISTS})
+        if (EXISTS ${check_exist_dir})
+            message("${check_exist_dir} exists, skipping download")
+            set(DOWNLOAD_ZIP 0)
+            break()
+        endif ()
+    endforeach(check_exist_dir)
+
     if (DOWNLOAD_ZIP)
         message("Downloading ${DOWNLOAD_OPENRCT2_DOWNLOAD_DIR}")
         file(DOWNLOAD

